### PR TITLE
Ruff life

### LINKED ISSUE #552: Replace Black and Isort with Ruff
ruff replaces both of these tools and is much much faster

I don't care about the actual format so much as long as it's consistent and I don't have to wait the tool to get the job done.

### PR DESCRIPTION
**Pull request by eevelweezel**
Setup pre-commit hooks for ruff & isort, remove CQ deps from pyproject.toml, and enable pre-commit.ci for the repo.

Closes #552 

@elmq0022 what is the workflow for pip-compile? I'd like to update the req.txt files to match.

Also, apologies for the giant diff... apparently, pre-commit.ci only works if everything passes the check.  I'm not sure I like that, but it does run the checks/fixes as part of CI. 

This PR includes a round of #noqa tags on auto-fixes ruff marked as potentially breaking.

### CODE REVIEW

**Comment by eevelweezel** [comment]
@elmq0022 please don't let this impede the tinymce work, I'll happily rebase if needed.

**Comment by elmq0022** [comment]
Yeah I would like to skip formatting migrations and other autogenerated files if possible.  

I don't know if it will make a mess when I squash migrations or not.  Probably okay though.

**Comment by elmq0022** [comment]
For pip compile create a .venv that has a matching python version (3.11.10) and install pip-tools in it. 

Update the pyproject.toml with whatever packages in the application sections. 

After that run the pip-compile commands found at the top of requirements.txt and the dev-requirements.txt.  

You can run something like pip-sync dev-requirements.txt to build to sync your .venv with the compiled requirements.   

I've just been running a .venv right in the project directory as it's pretty convenient to source it from there.  

The official docs are pretty good and quite short too.  

I'll need to add this to our readme and remove anything about pipenv.

**Comment by eevelweezel** [comment]
I'll update the pip-compile stuff in the next few days.  This PR can wait until after the tinymce/upgrade migrations are merged.

File: pyproject.toml
Comment (elmq0022) [inline pyproject.toml:1]: We should add all the ignore directories here plus our migrations directory.  

https://docs.astral.sh/ruff/configuration/

Some of the other configs look nice as well.  Like double quotes and linting.

**Comment by elmq0022** [comment]
> I'll update the pip-compile stuff in the next few days. This PR can wait until after the tinymce/upgrade migrations are merged.

Thanks @eevelweezel.  I'll merge everything this Saturday when I have a little more time available to check data migrations.

**Review by elmq0022 — APPROVED** [review]

File: pyproject.toml
Comment (eevelweezel) [reply pyproject.toml:1]: Sorry again for this, it's been a minute since I worked on anything with migrations and I completely forgot.  I stopped skipping tests as a matter of preference, but my typical project is way smaller than this one.

**Pull request closed without merging**
